Mobile Release v1.46.0

fluiddot opened the pull request:
## Description
Release 1.46.0 of the react-native-editor and Gutenberg-Mobile.

For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3078

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/native-mobile.md -->

fluiddot committed:
Release script: Update react-native-editor version to 1.46.0

fluiddot committed:
Release script: Update with changes from 'npm run core preios'

etoledom committed:
[RNMobile] Audio Player UI for audio block (#27467)

* Basics of Audio block working

* Add audio support to MediaUpload

* Add handling of file uploads and replace

* WPMediaLibrary support for Audio block

* Avoid removing media info on error state

* Linting

* Added an AUDIO file to the test requestMediaPickFromMediaLibrary func

* Fixed typo in ToolbarButton of Audio Block.

* Removed auto help behavior present on web that's not used on mobile.

* [Android] Wired the click of the Audio Media Library button.

* Added Audio media options for choosing audio file locally.

* [RNMobile] Audio Block: Proper caption field (#27689)

* Audio Player UI for audio block

* Add extension to styles import

* Show file name while loading and retry message on error

* Pass state props to audio player component

* Implement placeholder-ish player UI structure

* added styles for icon, title and subtitle

* added blue-wordpress as link color.

* added sizing based on design specs.

* Fixed object destructuring error.

* added icon styling state for upload in progress or upload failed.

* implemented error style and behavior.

* added styling for upload failed text.

* Override MediaUploadProgress styles

* Update filename extension spseration to handle two dots

* Update UI structure and styles

* Make retry message translateable

* Update snaphots

* Fix lint error

* Set initial file name

* Remove devOnly flag from audio block

* Increase tap target of button

* Add 1px between title and subtitle

* Align title, subtitle and button vertically

* On iOS use VideoPlayer to play audio files in-app

* Revert "On iOS use VideoPlayer to play audio files in-app"

This reverts commit b1eb8dd5d110b4243d3f71e0121bab92266151ec.

* [RNMobile] Audio Block -  Cancel and Retry Dialog (#28540)

Co-authored-by: Ceyhun Ozugur <ceyhunozugur@gmail.com>

* Rename button title

* Add pill-shaped background to button

* Add padding to title

* Fix file title being empty when selecting from media library

* Use safeDecodeURI

* On iOS use VideoPlayer to play audio files in-app

* Decrease button font size

Co-authored-by: Joel Dean <jdeanjj1000@gmail.com>

* Revert "Remove devOnly flag from audio block"

This reverts commit e6b5b6df7b28570fc6bfbe28239eff16a5e3039d.

* Add right padding to title container to fix error state

Co-authored-by: Joel Dean <jdeanjj1000@gmail.com>
Co-authored-by: Ceyhun Ozugur <ceyhunozugur@gmail.com>

lukewalczak committed:
[RNMobile] Correct condition in setting blockWidth (#28635)

dratwas committed:
[RNMobile] Fix height of the bottom-sheet after using back button (#28632)

* remove unused param and setHeight even if it is the same

ceyhun committed:
[RNMobile] Enable Audio Block (#28594)

Co-authored-by: Joel Dean <jdeanjj1000@gmail.com>

jd-alexander commented:
Hi @ceyhun @fluiddot for the release notes, I am assuming that once all the cherry-picking has been done, that will bring in all the changes of the release notes to that `unreleased` section of the CHANGELOG.md I realized that the #28635 and 28632 don't contain changes to the release notes. So will these be added here? I am checking about this to ensure I haven't missed any steps about it in the release process checklist. I also know that the `unreleased` is supposed to change to the current version being released as well so I am just noting that here to ensure I am following the process correctly.

fluiddot commented:
> Hi @ceyhun @fluiddot for the release notes, I am assuming that once all the cherry-picking has been done, that will bring in all the changes of the release notes to that `unreleased` section of the CHANGELOG.md. I realized that the #28635 and #28632 don't contain changes to the release notes. So will these be added here? I am checking about this to ensure I haven't missed any steps about it in the release process checklist. I also know that the `unreleased` is supposed to change to the current version being released as well so I am just noting that here to ensure I am following the process correctly.

As far as I checked in recent releases, the [`CHANGELOG.md`](https://github.com/WordPress/gutenberg/blob/master/packages/react-native-editor/CHANGELOG.md) and [`RELEASE-NOTES.txt`](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) files are updated manually when cutting so now that all the cherry picks have been done, I'll review and update them.

Regarding #28635 and #28632 PRs, I'll ask the authors if they consider that their changes should be also added there.

Thanks @jd-alexander for checking this!

fluiddot committed:
Update react-native-editor CHANGELOG

fluiddot committed:
Update unsupported block test data for UI tests (#28649)

The unsupported block now used is jetpack/gif since Audio block is now supported.

ceyhun commented on packages/react-native-aztec/package.json:
```diff
@@ -1,43 +1,43 @@
-		"clean-watchman": "command -v watchman >/dev/null 2>&1 && watchman watch-del-all;",
-		"clean:install": "npm run clean && npm install"
-	}
+  "name": "@wordpress/react-native-aztec",
```
Gutenberg seems to use tabs for all the package.json files and this conversion to spaces seems to happen because of a missing flag in `jq`, and will be converted back in a future PR like this: https://github.com/WordPress/gutenberg/pull/28278

Created a PR in release toolkit to fix this: https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/pull/20

Maybe we can convert back all the modified package.json files to tabs here as well.

fluiddot replied:
Ok good idea, I'll revert them back 👍 .

ceyhun commented on packages/react-native-editor/CHANGELOG.md:
```diff
@@ -10,12 +10,10 @@ For each user feature we should also add a importance categorization label  to i
 -->
 
 ## Unreleased
```
We can maybe move these under `1.46.0` still keeping the `Unreleased`

fluiddot replied:
Yeah, I was thinking to do it after we finish the testing but I'm ok to do it now too 😉 .

fluiddot committed:
Revert indentation of modified package.json files

fluiddot committed:
Update react-native-editor CHANGELOG

lukewalczak committed:
[RNMobile] Move custom indicator to be rendered above the colors (#28698)

lukewalczak committed:
[RNMobile] Move back disabled style for button in File block (#28704)

ceyhun reviewed (approved):
:shipit:

fluiddot committed:
Merge branch 'master' into rnmobile/release_1.46.0

# Conflicts:
#	packages/react-native-editor/CHANGELOG.md
